core/src/main/java/com/redhat/lightblue/client/request/data/LiteralDataRequest.java

```diff
@@ -12,29 +15,46 @@
  */
 public class LiteralDataRequest extends AbstractLightblueDataRequest {
 
-    private final String body;
+    private final JsonNode body;
     private final HttpMethod httpMethod;
     private final String operationPathParam;
 
+    @Deprecated
     public LiteralDataRequest(String body, HttpMethod httpMethod, String operationalPathParam) {
         super();
-        this.body = body;
+        this.body = JSON.toJsonNode(body);
         this.httpMethod = httpMethod;
         this.operationPathParam = operationalPathParam;
     }
 
+    @Deprecated
     public LiteralDataRequest(String entityName, String entityVersion, String body, HttpMethod httpMethod, String operationalPathParam) {
         super(entityName, entityVersion);
+        this.body = JSON.toJsonNode(body);
+        this.httpMethod = httpMethod;
+        this.operationPathParam = operationalPathParam;
+    }
+
+    
+    public LiteralDataRequest(JsonNode body, HttpMethod httpMethod, String operationalPathParam) {
+        super();
+        this.body = body;
+        this.httpMethod = httpMethod;
+        this.operationPathParam = operationalPathParam;
+    }
+
+    public LiteralDataRequest(String entityName, String entityVersion, JsonNode body, HttpMethod httpMethod, String operationalPathParam) {
+        super(entityName, entityVersion);
         this.body = body;
         this.httpMethod = httpMethod;
         this.operationPathParam = operationalPathParam;
     }
 
     @Override
-    public String getBody() {
+    public JsonNode getBodyJson() {
```

Comment:
This is a potentially breaking change. I have no objections, but if we are going to force this, does it make just as much sense to rip out the other depreciated classes/enums and just rip the bandage off?